lib/mix/tasks/phoenix.gen.channel.ex

```diff
@@ -25,6 +25,8 @@ defmodule Mix.Tasks.Phoenix.Gen.Channel do
 
     binding = binding ++ [plural: plural]
 
+    Mix.Phoenix.check_module_name_availability!(binding[:module])
```

Comment:
@josevalim is this what you had in mind?  Thank you very much for your feedback.